crates/sp-lightclient/src/lib.rs

```diff
@@ -221,6 +229,12 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
         };
 
         store.store_header(header_ext, is_best_header);
+
+        // finalize and prune forks if the chain has progressed
+        if is_best_header {
+            Self::finalize_header_at_k_depth(store)?;
```

Comment:
It might be helpful to initialize `HeaderImporter` with `Store` since it is used in multiple methods and isn't really supposed to be used with different stores for different calls, here you're calling a static method, but could have been a regular one with one less explicit argument.